assets/js/testScript.js

```diff
@@ -0,0 +1,488 @@
+    if ($playPos.top !== 590) { $player.css('top', `590px`) }
+  })
+
+  function bulletCollisionCheck (obj1, obj2) { //bullet collision check
+    var red = obj1.position()
+    var redX = red.left
+    var blue = obj2.jTarget.position()
+    var blueX = blue.left
+    var redY = red.top
+    var blueY = blue.top
+    if (blueX < redX + obj1.width() &&
+        blueX + obj2.jTarget.width() > redX &&
+        blueY < redX + obj1.height() &&
+        obj2.jTarget.height() + blueY > redY) {
+      obj2.hitPoints -= 1
+      $bullet.css('left', $playPos.left)
+      $bullet.css('top', '620px')
+      if (obj2.hitPoints === 0) {
+        obj2.jTarget.remove()
+        if (obj2.bosscheck === false) {
+          mobArray.splice(key, 1)
+          mobsOnScreen--
+          levelOneMobCount--
+        } else if (obj2.bosscheck) {
+          bossArray.splice(key, 1)
+          bossDead = true
+        }
+      }
+      if (mobsOnScreen === 0 && levelOneMobCount === 0) { //if mobs are dead, spawn boss
+        bossLevel = true
+      }
+      if (bossLevel === true && bossDead === true) {   //if boss is dead, end game
+        gameEnd = true
+      }
+      return true
+    } else { return false }
+  }
+
+  function playerMobCollisionCheck (obj1, obj2) {     //player - mob collision check
+    var red = obj1.position()
+    var redX = red.left
+    var blue = obj2.jTarget.position()
+    var blueX = blue.left
+    var redY = red.top
+    var blueY = blue.top
+
+    if (blueX < redX + $player.width() &&
+         blueX + obj2.jTarget.width() > redX &&
+         blueY < redX + $player.height() &&
+         obj2.jTarget.height() + blueY > redY) {
+      console.log('PLAYER HEALTH REDUCED')
+      playerHealth -= obj2.damage
+      var currentHpWidth = $('.hpBar').width()
+      $('.hpBar').css('width', `${currentHpWidth - obj2.damage / 5}px`)
+      $hpBar.text(`${playerHealth}/1000`)
+      if (playerHealth === 0) {
+        gameEnd = true
+      }
+    } else {
+      return false
+    }
+  }
+
+  function fireBullet () {      //bullet control functiom
+    $playPos = $player.position()
+    $mobPos = $mob.position()
+    $bullet.show()
+    $bulletLoc = $bullet.position()
+    $bulletExact = $bulletLoc.left
+    $bulletTop = $playPos.top
+    $bulletLeft = $playPos.left
+    $bullet.css('left', $playPos.left)
+    if (fireDirection === 'right') {
+      $bullet.css('left', `${$bulletExact + (1600 - $bulletExact)}px`)
+    } else if (fireDirection === 'left') {
+      $bullet.css('left', `${$bulletExact - (1600 - $bulletExact)}px`)
+    }
+    if ($bulletExact <= 200 || $bulletExact >= 1600) {  //reset when it moves out of screen
+      $bullet.css('left', $playPos.left)
+      console.log('bullet', $bulletExact)
+    }
+    setTimeout(function(){$bullet.hide()},1000)
+  }
+  function skillOne () {    // small damage skill
+    if (mana >= 10) {
+      mana -= 10
+      $mpBar.text(`${mana}/200`)
+      $mpBar.css('width', `${$mpBar.width() - 10}px`)
```

Comment:
Why not make the position = $player.position().left ?

Reply:
its to reduce the width of the blue bar as the mana is used up, doesn't have any link to player position